cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceTaskContext.java

```diff
@@ -41,7 +42,7 @@
  */
 @Beta
 public interface MapReduceTaskContext<KEYOUT, VALUEOUT> extends SchedulableProgramContext, RuntimeContext,
-  DatasetContext, ServiceDiscoverer, PluginContext, TaskLocalizationContext, SecureStore {
+  DatasetContext, ServiceDiscoverer, PluginContext, TaskLocalizationContext, SecureStore, MetadataReaderContext {
```

Comment:
should this be in the task context? That is, every task can read meta data? That can cause a huge load of requests... I would like to encourage developers to call it in the driver and serialize it to all tasks.